src/annotator/range-util.js

```diff
@@ -114,7 +114,11 @@ export function selectionFocusRect(selection) {
   if (selection.isCollapsed) {
     return null;
   }
-  const textBoxes = getTextBoundingBoxes(selection.getRangeAt(0));
+  const range = selection.getRangeAt(0);
+  if (range.toString().trim() === '') {
```

Comment:
I have seen cases with extremely large selections (eg. select a large proportion of the text on the HTML spec page) where `Range.toString` can be slow. I'd guess that the existing `getTextBoundingBoxes` is probably already much slower in that case.